### DISCUSSION TITLE
Audio Codec

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#51] [Ideas]
What would be the needed function to support other audio codec like mp3 or flac ?

https://github.com/xiph/flac/tree/master/src/libFLAC
https://www.mpg123.de/trunk/src/libmpg123/

**Comment by VendelinSlezak** [comment]
For any audio format we need code that uncompress raw PCM data from file. I would really like to have support for mp3, and I already studied something about it and I understand some basic concepts and I know that for first part of decompressing mp3 - huffman decoding - have BleskOS already some general code. But I am not exactly sure how to extract huffman tables from it and there are like four or maybe five more steps to achieve PCM data, so now I am not even close to be able to write full driver.

**Comment by defdefred** [comment]
Ok and do you want to have the whole decoded wav in ram (could be a issue with big file) or rather decoded like a stream?

**Comment by VendelinSlezak** [comment]
I do not have exact architecture idea yet, but I will go with stream, because raw PCM data are big.

**Comment by VendelinSlezak** [comment]
I added support for MP3 in Update 31 and I create model for decoding it as stream, so no additional memory than memory for MP3 file itself is needed.
